#!/usr/bin/env python

# Define your cpp namespace here, due to the fact that hyphen and dashes is not used for variable name, use underscore instead.  
CPPNAMESPACE="ros_dynamic_reconfigure"

from dynamic_reconfigure.parameter_generator_catkin import *


# Parameter generator handle
gen = ParameterGenerator()

# Populate drop-down menu content
# Name | Type | Value | Description
drop_down_menu = gen.enum([
    gen.const("default_value", int_t, -1, "Not used"),
    gen.const("value_zero", int_t, 0, "This is value zero"),
    gen.const("value_one", int_t, 1, "This is value one")],
    "Drop_down Menu"
)

# Add drop-down menu to window
# Name | Type | Level | Description | Default | Min | Max | Values
gen.add("drop_down_menu", int_t, 0, "Drop_down Menu", -1, -1, 1, edit_method=drop_down_menu)

# Add boolean option to window
# Name | Type | Level | Description | Default | Min | Max | Values
gen.add("true_or_false", bool_t, 0, "True or False", True)

# Add double option to window
# Name | Type | Level | Description | Default | Min | Max | Values
gen.add("slider_points", double_t, 0, "Slider", 0.7, 0, 1)

# Exit parameter generator
# Package Name | Node Name (used to generate documentation only) | cfg File Name
exit(gen.generate(CPPNAMESPACE, "RosDynamicReconfigure", "RosDynamicReconfigure"))
